src/main/java/com/github/mc1arke/sonarqube/plugin/scanner/autoconfiguration/GithubActionsAutoConfigurer.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2022 Michael Clarke
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ */
+package com.github.mc1arke.sonarqube.plugin.scanner.autoconfiguration;
+
+import com.github.mc1arke.sonarqube.plugin.scanner.BranchAutoConfigurer;
+import com.github.mc1arke.sonarqube.plugin.scanner.BranchConfigurationFactory;
+import org.apache.commons.lang.StringUtils;
+import org.sonar.api.utils.System2;
+import org.sonar.scanner.scan.branch.BranchConfiguration;
+import org.sonar.scanner.scan.branch.ProjectBranches;
+
+import java.util.Optional;
+
+public class GithubActionsAutoConfigurer implements BranchAutoConfigurer {
+
+    private static final String PULL_REQUEST_REF_PREFIX = "refs/pull/";
+    private static final String PULL_REQUEST_REF_POSTFIX = "/merge";
+
+    private final BranchConfigurationFactory branchConfigurationFactory;
+
+    public GithubActionsAutoConfigurer(BranchConfigurationFactory branchConfigurationFactory) {
+        this.branchConfigurationFactory = branchConfigurationFactory;
+    }
+
+    @Override
+    public Optional<BranchConfiguration> detectConfiguration(System2 system, ProjectBranches projectBranches) {
+        if (!Boolean.parseBoolean(system.envVariable("GITHUB_ACTIONS"))) {
+            return Optional.empty();
+        }
+
+        String branch = system.envVariable("GITHUB_REF");
+        if (StringUtils.isEmpty(branch)) {
+            return Optional.empty();
+        }
+
+        String sourceBranch = system.envVariable("GITHUB_HEAD_REF");
+        if (branch.startsWith(PULL_REQUEST_REF_PREFIX) && branch.endsWith(PULL_REQUEST_REF_POSTFIX) && StringUtils.isNotEmpty(sourceBranch)) {
+            String key = branch.substring(PULL_REQUEST_REF_PREFIX.length(), branch.length() - PULL_REQUEST_REF_POSTFIX.length());
+            String targetBranch = system.envVariable("GITHUB_BASE_REF");
+            return Optional.of(branchConfigurationFactory.createPullRequestConfiguration(key, sourceBranch, targetBranch, projectBranches));
+        } else {
+            return Optional.of(branchConfigurationFactory.createBranchConfiguration(branch.substring(11), projectBranches));
```

Comment:
You can use `GITHUB_REF_NAME`:
```suggestion
        String branch = system.envVariable("GITHUB_REF_NAME");
        if (StringUtils.isEmpty(branch)) {
            return Optional.empty();
        }

        String sourceBranch = system.envVariable("GITHUB_HEAD_REF");
        if (branch.startsWith(PULL_REQUEST_REF_PREFIX) && branch.endsWith(PULL_REQUEST_REF_POSTFIX) && StringUtils.isNotEmpty(sourceBranch)) {
            String key = branch.substring(PULL_REQUEST_REF_PREFIX.length(), branch.length() - PULL_REQUEST_REF_POSTFIX.length());
            String targetBranch = system.envVariable("GITHUB_BASE_REF");
            return Optional.of(branchConfigurationFactory.createPullRequestConfiguration(key, sourceBranch, targetBranch, projectBranches));
        } else {
            return Optional.of(branchConfigurationFactory.createBranchConfiguration(branch, projectBranches));
```

Reply:
I can't use `GITHUB_REF_NAME` at the point you've suggested as line in 53 depends on knowing the full ref value. It would be possible to use the `GITHUB_REF_NAME` environment variable in line 54 and 58, although given we know the format of the ref at that point I'm not sure if this provides a huge amount of value.